backend/cfg/vectorize.ml

```diff
@@ -134,7 +138,327 @@ module Dependency_graph = struct
+          | Global, Local -> -1
+          | Local, Global -> 1)
+        | Ibased _, _ -> -1
+        | _, Ibased _ -> 1
+        | Iindexed n1, Iindexed n2 ->
+          let arg0_1 = Array.get args_1 0 in
+          let arg0_2 = Array.get args_2 0 in
+          let arg0_compare = arg0_1.stamp - arg0_2.stamp in
+          if arg0_compare = 0 then n1 - n2 else arg0_compare
+        | Iindexed _, _ -> -1
+        | _, Iindexed _ -> 1
+        | Iindexed2 n1, Iindexed2 n2 ->
+          let arg0_1 = Array.get args_1 0 in
+          let arg0_2 = Array.get args_2 0 in
+          let arg0_compare = arg0_1.stamp - arg0_2.stamp in
+          if arg0_compare = 0
+          then
+            let arg1_1 = Array.get args_1 1 in
+            let arg1_2 = Array.get args_2 1 in
+            let arg1_compare = arg1_1.stamp - arg1_2.stamp in
+            if arg1_compare = 0 then n1 - n2 else arg1_compare
+          else arg0_compare
+        | Iindexed2 _, _ -> -1
+        | _, Iindexed2 _ -> 1
+        | Iscaled (scale1, n1), Iscaled (scale2, n2) ->
+          let arg0_1 = Array.get args_1 0 in
+          let arg0_2 = Array.get args_2 0 in
+          let arg0_compare = arg0_1.stamp - arg0_2.stamp in
+          if arg0_compare = 0
+          then
+            let scale_compare = scale1 - scale2 in
+            if scale_compare = 0 then n1 - n2 else scale_compare
+          else arg0_compare
+        | Iscaled _, _ -> -1
+        | _, Iscaled _ -> 1
+        | Iindexed2scaled (scale1, n1), Iindexed2scaled (scale2, n2) ->
+          let arg0_1 = Array.get args_1 0 in
+          let arg0_2 = Array.get args_2 0 in
+          let arg0_compare = arg0_1.stamp - arg0_2.stamp in
+          if arg0_compare = 0
+          then
+            let arg1_1 = Array.get args_1 1 in
+            let arg1_2 = Array.get args_2 1 in
+            let arg1_compare = arg1_1.stamp - arg1_2.stamp in
+            if arg1_compare = 0
+            then
+              let scale_compare = scale1 - scale2 in
+              if scale_compare = 0 then n1 - n2 else scale_compare
+            else arg1_compare
+          else arg0_compare
+      in
+      if compare_addressing_modes = 0
+      then Instruction.id t1.instruction - Instruction.id t2.instruction
+      else compare_addressing_modes
+
+    let offset_of (t1 : t) (t2 : t) =
+      let addressing_mode_1 = t1.addressing_mode in
+      let addressing_mode_2 = t2.addressing_mode in
+      let args_1 = args t1 in
+      let args_2 = args t2 in
+      match addressing_mode_1, addressing_mode_2 with
+      | Ibased (symbol1, global1, n1), Ibased (symbol2, global2, n2) -> (
+        match global1, global2 with
+        | Global, Global | Local, Local ->
+          if symbol1 = symbol2 then Some (n2 - n1) else None
+        | Global, Local | Local, Global -> None)
+      | Iindexed n1, Iindexed n2 ->
+        let arg0_1 = Array.get args_1 0 in
+        let arg0_2 = Array.get args_2 0 in
+        let arg0_compare = arg0_1.stamp - arg0_2.stamp in
```

Comment:
`Reg.stamp`  is an implementation detail that shouldn't be used outside of `Reg`. Please use `Reg.same` . You can also have a helper function `same_args: Reg.t array -> Reg.t array -> bool` to avoid duplication.

Reply:
same gives a boolean value and I would like to have a comparison

Reply:
I added a compare function to reg.ml